pkg/cloudprovider/provider/anexia/provider.go

```diff
@@ -278,10 +282,19 @@ func (p *provider) Cleanup(machine *v1alpha1.Machine, _ *cloudprovidertypes.Prov
 	err = apiClient.VSphere().Provisioning().VM().Deprovision(ctx, status.InstanceID, false)
 	if err != nil {
 		var respErr *anxclient.ResponseError
-		if errors.As(err, &respErr) && respErr.ErrorData.Code == http.StatusNotFound {
-			return true, nil
+		// Only error if the error was not "not found"
+		if !(errors.As(err, &respErr) && respErr.ErrorData.Code == http.StatusNotFound) {
```

Comment:
why did you add `!` to the condition?

Reply:
I inverted it because we have to continue with freeing the ip afterwards.
The `err was 404` path is now a fallthrough to the `free ip` codepath.